provider/maas/environ.go

```diff
@@ -614,38 +614,38 @@ func getCapabilities(client *gomaasapi.MAASObject, serverURL string) (set.String
 	for a := shortAttempt.Start(); a.Next(); {
 		version := client.GetSubObject("version/")
 		result, err = version.CallGet("", nil)
-		if err != nil {
-			if err, ok := errors.Cause(err).(gomaasapi.ServerError); ok && err.StatusCode == 404 {
-				message := "could not connect to MAAS controller - check the endpoint is correct"
-				trimmedURL := strings.TrimRight(serverURL, "/")
-				if !strings.HasSuffix(trimmedURL, "/MAAS") {
-					message += " (it normally ends with /MAAS)"
-				}
-				return caps, errors.NewNotSupported(nil, message)
-			}
-		} else {
+		if err == nil {
 			break
 		}
+		if err, ok := errors.Cause(err).(gomaasapi.ServerError); ok && err.StatusCode == 404 {
+			message := "could not connect to MAAS controller - check the endpoint is correct"
+			trimmedURL := strings.TrimRight(serverURL, "/")
+			if !strings.HasSuffix(trimmedURL, "/MAAS") {
+				message += " (it normally ends with /MAAS)"
+			}
+			return caps, errors.NewNotSupported(nil, message)
+		}
 	}
 	if err != nil {
 		return caps, err
 	}
 	info, err := result.GetMap()
 	if err != nil {
-		return caps, err
+		// invalid data of some sort, probably not a MAAS server.
+		return caps, errors.New("failed to get expected data from server")
 	}
 	capsObj, ok := info["capabilities"]
 	if !ok {
 		return caps, fmt.Errorf("MAAS does not report capabilities")
 	}
 	items, err := capsObj.GetArray()
 	if err != nil {
-		return caps, err
+		return caps, errors.New("failed to get expected data from server")
```

Comment:
We don't want to completely discard the underlying error. `errors.Wrap(err, errors.New("failed to get expected data from server"))`.

I thought we were also going to log this to Debug?